### PR TITLE
Chart version bump fix

### PR DESCRIPTION
**Pull request by RafalKorepta**
Align all appVersion string with artifacthub.io/images annotation.

Fix operator values.yaml to work with `bump_chart_version.sh` script.
Currently nightly build is failing with the following error:
```
2023/09/15 01:26:17 Error querying repository tags: unable to open repository 'docker.redpanda.com/redpandadata/redpanda-operator
docker.redpanda.com/redpandadata/configurator': invalid reference: invalid repository
```

The values.yaml file for operator has 2 repository values which confuses `bump_chart_version.sh` script.

The `docker-tag-list` program receives in the REPO variable the following content:
```
docker.redpanda.com/redpandadata/redpanda-operator
docker.redpanda.com/redpandadata/configurator
```

This change overcome this issue.

P.S. New version of the Operator will not use configurator container.

REF
https://github.com/redpanda-data/helm-charts/actions/runs/6192749166/job/16813278284#step:6:7

Make new replacement of any occurrence of old version into new version in `Chart.yaml` file.

### CODE REVIEW

File: charts/operator/Chart.yaml [hunk 1]
```diff
@@ -34,9 +34,9 @@ annotations:
       url: https://helm.sh/docs/intro/install/
   artifacthub.io/images: |
     - name: redpanda-operator
-      image: docker.redpanda.com/redpandadata/redpanda-operator:v23.2.5
```
Comment (alejandroEsc) [inline charts/operator/Chart.yaml:37]: should we use 23.2.8 instead?

File: .github/bump_chart_versions.sh [hunk 2]
```diff
@@ -40,4 +40,5 @@ fi
 echo "update ${CHART} appVersion from ${APPVERSION} to ${LATEST}"
 bump_patch_version
 sed -i "s/^version: .*$/version: ${NEW_VERSION}/" "${CHARTFILE}"
-sed -i "s/^appVersion: .*$/appVersion: ${LATEST}/" "${CHARTFILE}"
\ No newline at end of file
+sed -i "s/^appVersion: .*$/appVersion: ${LATEST}/" "${CHARTFILE}"
+sed -i "s/${APPVERSION}/${LATEST}/" "${CHARTFILE}"
```
Comment (alejandroEsc) [inline .github/bump_chart_versions.sh:44]: probably will need an extra blank line here,

File: charts/operator/templates/deployment.yaml [hunk 3]
```diff
@@ -48,7 +48,7 @@ spec:
         - --metrics-bind-address=127.0.0.1:8080
         - --leader-elect
         - --configurator-tag={{ include "configurator.tag" . }}
-        - --configurator-base-image={{ .Values.configurator.repository }}
+        - --configurator-base-image={{ .Values.configurator.containerRegistry }}
```
Comment (alejandroEsc) [inline charts/operator/templates/deployment.yaml:51]: why this change?

**Review by alejandroEsc — COMMENTED** [review]
A few questions, not big.

File: .github/bump_chart_versions.sh (hunk 2 above)
Comment (joejulian) [inline .github/bump_chart_versions.sh:44]: I was concerned that there could be a possibility that `$APPVERSION` might match some other string, like if we start adding release notes to the artifacthub metadata. It's also not quoting the dots so they'll be wildcards in this match.

I was thinking about maybe using yq (or gojq) to do this, roughly like:
```
IMAGES=$(yq -r '.annotations["artifacthub.io/images"]' charts/console/Chart.yaml)
NEWIMAGES=$(yq -c --arg v $LATEST '(. | select(.name == "redpanda") | .image) |= "docker.redpanda.com/redpandadata/console:"+$v' <<< $IMAGES)
yq -Y --argjson v $NEWIMAGES '.annotations["artifacthub.io/images"] = $v' charts/console/Chart.yaml > temp
mv temp charts/console/Chart.yaml
```

File: charts/operator/templates/deployment.yaml (hunk 3 above)
Comment (joejulian) [reply charts/operator/templates/deployment.yaml:51]: I was wondering that, too.

File: charts/operator/values.yaml [hunk 4]
```diff
@@ -24,7 +24,7 @@ image:
 
 configurator:
   # configurator.repository -- Repository that Redpanda configurator image is available
-  repository: docker.redpanda.com/redpandadata/configurator
+  containerRegistry: docker.redpanda.com/redpandadata/configurator
```
Comment (joejulian) [inline charts/operator/values.yaml:27]: This is a breaking change. Is it needed?

**Review by joejulian — CHANGES_REQUESTED** [review]
blocking on the breaking change

File: charts/operator/values.yaml (hunk 4 above)
Comment (RafalKorepta) [reply charts/operator/values.yaml:27]: I'm not that proficient in AWK to get the correct repository (container registry to be specific) from helm values yaml only if it comes from specific place. That place in values.yaml is not the same across different helm charts. The comment as far as see is different.

File: charts/operator/Chart.yaml (hunk 1 above)
Comment (RafalKorepta) [reply charts/operator/Chart.yaml:37]: I wanted to merge this PR to run nightly and create PR from that run.

File: .github/bump_chart_versions.sh (hunk 2 above)
Comment (RafalKorepta) [reply .github/bump_chart_versions.sh:44]: Does shell need empty line at the end of the file?

File: charts/operator/templates/deployment.yaml (hunk 3 above)
Comment (RafalKorepta) [reply charts/operator/templates/deployment.yaml:51]: I highly recommend reading my commit message. I will put the whole reasoning into PR cover letter

```
operator: Change configurator container registry values yaml
Currently nightly build is failing with the following error:
```
2023/09/15 01:26:17 Error querying repository tags: unable to open repository 'docker.redpanda.com/redpandadata/redpanda-operator
docker.redpanda.com/redpandadata/configurator': invalid reference: invalid repository
```

The values.yaml file for operator has 2 repository values which confuses `bump_chart_version.sh` script.

The `docker-tag-list` program receives in the REPO variable the following content:
```
docker.redpanda.com/redpandadata/redpanda-operator
docker.redpanda.com/redpandadata/configurator
```

This change overcome this issue.

P.S. New version of the Operator will not use configurator container.

REF
https://github.com/redpanda-data/helm-charts/actions/runs/6192749166/job/16813278284#step:6:7
```

**Comment by RafalKorepta** [comment]
The https://github.com/redpanda-data/helm-charts/pull/842 fixed operator issue
